### PR TITLE
Add login and password through secret workspace

### PR DESCRIPTION
**Pull request by GijsvanDulmen**
# Changes

Add a possibility to optionally mount a workspace with sonarqube credentials.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [X] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [X] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [X] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [X] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [X] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [X] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [X] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [X] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

**Comment by GijsvanDulmen** [comment]
/remove-lifecycle stale

Only needs approval

File: task/sonarqube-scanner/0.3/sonarqube-scanner.yaml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+          touch sonar-project.properties
+          echo "sonar.projectKey=$(params.SONAR_PROJECT_KEY)" >> sonar-project.properties
+          echo "sonar.host.url=$(params.SONAR_HOST_URL)" >> sonar-project.properties
+          echo "sonar.sources=." >> sonar-project.properties
+        fi
+
+        if [ "$(workspaces.sonar-credentials.bound)" == "true" ] ; then
+          if [[ -f $(workspaces.sonar-credentials.path)/$(params.SONAR_PASSWORD_KEY) ]]; then
+            SONAR_PASSWORD=`cat $(workspaces.sonar-credentials.path)/$(params.SONAR_PASSWORD_KEY)`
+            replaceValues $(workspaces.source.path)/sonar-project.properties sonar.password ${SONAR_PASSWORD}
```
[the remark this answers is not in the document]
Comment (GijsvanDulmen) [reply task/sonarqube-scanner/0.3/sonarqube-scanner.yaml]: I added the quotes!

**Comment by GijsvanDulmen** [comment]
@afrittoli Could you check again please? Thanks!

**Comment by GijsvanDulmen** [comment]
0.3 already contains the functionality now ! 🙃

**Comment by GijsvanDulmen** [comment]
Or maybe not. But I'll make sure there is a new PR.

**Comment by chmouel** [comment]
ah cool, feel free to ping me directly!

**Pull request closed without merging**